draft-ietf-intarea-proxy-config.md

```diff
@@ -233,38 +233,63 @@ content-length = 222
-that are accessible using the proxy. If a hostname is not included in the enumerated
-zones, then a client SHOULD assume that the hostname will not be accessible through the
-proxy. If a hostname is included in the `excludedDomains` array, then the client SHOULD NOT
-access it through the proxy.
+as described in {{proxy-pvd}}, entries in the `matchDomains` array indicate specific hosts
+and zones that are accessible using the proxy. If a hostname is neither matching an specific
+entry nor included in the enumerated zones, then a client SHOULD assume that the hostname
+will not be accessible through the proxy. If a hostname is included in the `excludedDomains`
+array, then the client SHOULD NOT access it through the proxy.
 
 Entries listed in `matchDomains` MUST NOT expand the set of domains that a client is
 willing to send to a particular proxy. The list can only narrow the list of domains
 that the client is willing to send through the proxy. For example, if the client
-has a local policy to only send requests for "example.com" to a proxy
+has a local policy to only send requests for "*.example.com" to a proxy
 "proxy.example.com", and the `matchDomains` array contains "internal.example.com" and
 "other.company.com", the client would end up only proxying "internal.example.com"
 through the proxy.
 
+A wildcard prefix (`*.`) is used to indicate matching entire domains or subdomains instead of specific hostnames. Note
+that this can be used to match multiple levels of subdomains. For example "*.example.com"
+matches "internal.example.com" as well as "www.public.example.com".
+
+Entries in `matchSubnets` correspond to IP addresses and subnets that are available through the
+proxy, while entries in `excludeSubnets` define IP addresses and subnets that SHOULD NOT be used
+with the proxy. Subnet-based destination information SHOULD only be used when
+applications are communicating with destinations identified by only an IP address and not a hostname.
+
+`matchPorts` in a list of strings that can be used to instruct the client that only specific destination
+TCP or UDP ports are accessible through the proxy. The list may contain individual port numbers 
+(such as "80") or inclusive ranges of ports. For example "1024-2048" matches all ports from 1024 
+to 2048, including the 1024 and 1028.
+
+Note that clients with limited resources MAY not be able to process and utilize all
+entries of an excessively long list. In the case if the provided list is too long for a given
+client, it SHOULD process as many records from the beginning of `matchDomains`, `matchSubnets` and
+`matchPorts` lists. These lists SHOULD be sorted with the most important elements placed at
+the start of the list. If a client cannot consume all the entries in `excludedDomains` or
+`excludedIPs` it SHOULD NOT use given proxy configuration to avoid sending traffic that the proxy
+cannot process.
+
```

Comment:
I'm a bit concerned about this section so would prefer to remove it from the first cut of adding this support — let's discuss limited clients more!
```suggestion
```